tests/sync/app.cpp

```diff
@@ -824,6 +827,368 @@ TEST_CASE("app: link_user integration", "[sync][app]") {
+    auto dog_document2 = "{\"name\":\"fido\", \"breed\":\"french bulldog\"}";
+
+    auto email = util::format("realm_tests_do_autoverify%1@%2.com", random_string(10), random_string(10));
+    auto password = random_string(10);
+    
+    app.provider_client<App::UsernamePasswordProviderClient>()
+    .register_email(email,
+                    password,
+                    [&](Optional<app::AppError> error) {
+                        CHECK(!error);
+                    });
+    
+    app.log_in_with_credentials(realm::app::AppCredentials::username_password(email, password),
+                                [&](std::shared_ptr<realm::SyncUser> user, Optional<app::AppError> error) {
+                                    REQUIRE(user);
+                                    CHECK(!error);
+                                });
+    
+    collection.delete_many(dog_document, [&](uint64_t, Optional<app::AppError> error) {
+        CHECK(!error);
+    });
+    
+    
+    collection.delete_many(dog_document2, [&](uint64_t, Optional<app::AppError> error) {
+        CHECK(!error);
+    });
+    
+    SECTION("insert") {
+        
+        bool processed = false;
+        std::string dog_object_id;
+        
+        collection.insert_one("{\"bad\" : \"value\"}",
+                              [&](std::string, Optional<app::AppError> error) {
+            CHECK(error);
+        });
+        
+        collection.insert_one(dog_document,
+                              [&](std::string document_json, Optional<app::AppError> error) {
+            CHECK(!error);
+            auto json = nlohmann::json::parse(document_json);
+            auto object_id = json.at("insertedId").at("$oid").get<std::string>();
+            dog_object_id = object_id;
+        });
+        
+        auto documents = std::vector<std::string>();
+        documents.push_back(dog_document2);
+        
+        collection.insert_many(documents,
+                               [&](std::vector<std::string> inserted_docs,
+                                   Optional<app::AppError> error) {
+            CHECK(!error);
+            CHECK(inserted_docs.size() == 1);
+            processed = true;
+        });
+        
+        CHECK(processed);
+    }
+    
+    SECTION("find") {
+        bool processed = false;
+        
+        collection.find(dog_document,
+                        [&](std::string document_json, Optional<app::AppError> error) {
+            CHECK(!error);
+            auto json = nlohmann::json::parse(document_json);
+            CHECK(json.is_array());
+        });
+        
+        collection.find_one(dog_document,
+                            [&](Optional<std::string> document_json, Optional<app::AppError> error) {
+            CHECK(!error);
+            CHECK(!document_json);
+        });
+        
+        collection.insert_one(dog_document,
+                              [&](std::string, Optional<app::AppError> error) {
+            CHECK(!error);
```

Comment:
check that the string response contains a value

Reply:
now checking for an object id